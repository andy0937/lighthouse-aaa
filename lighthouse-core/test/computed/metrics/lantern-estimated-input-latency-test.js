/**
 * @license Copyright 2018 Google Inc. All Rights Reserved.
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
 */
'use strict';

const LanternEstimatedInputLatency = require('../../../computed/metrics/lantern-estimated-input-latency.js'); // eslint-disable-line max-len
const trace = require('../../fixtures/traces/progressive-app-m60.json');
const devtoolsLog = require('../../fixtures/traces/progressive-app-m60.devtools.log.json');

/* eslint-env jest */

describe('Metrics: Lantern EIL', () => {
  it('should compute a simulated value', async () => {
    const settings = {throttlingMethod: 'simulate'};
    const data = {trace, devtoolsLog, settings};
    const context = {settings, computedCache: new Map()};
    const result = await LanternEstimatedInputLatency.request(data, context);

    expect({
      timing: Math.round(result.timing),
      optimistic: Math.round(result.optimisticEstimate.timeInMs),
      pessimistic: Math.round(result.pessimisticEstimate.timeInMs),
    }).toMatchSnapshot();
  });
});
