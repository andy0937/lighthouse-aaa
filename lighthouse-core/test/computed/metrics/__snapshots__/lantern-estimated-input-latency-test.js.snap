// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics: Lantern EIL should compute a simulated value 1`] = `
Object {
  "optimistic": 93,
  "pessimistic": 101,
  "timing": 78,
}
`;
